### PR TITLE
Add dvm to crab

### PR DESCRIPTION
**Pull request by AsceticBear**
This PR only deals with the tasks that to be completed to add dvm into crab network. As for the crab version upgrade,  I will take a separate PR after this modification is approved.

### CODE REVIEW

File: runtime/crab/src/lib.rs [hunk 1]
```diff
@@ -936,6 +947,56 @@ impl darwinia_crab_issuing::Trait for Runtime {
...
+pub struct FixedGasPrice;
+impl FeeCalculator for FixedGasPrice {
+	fn min_gas_price() -> U256 {
+		// Gas price is always one token per gas.
+		0.into()
+	}
+}
+
+parameter_types! {
+	pub const ChainId: u64 = 43;
```
Comment (AsceticBear) [inline runtime/crab/src/lib.rs:960]: Need to check this ChainId again.

Comment (HackFisher) [reply runtime/crab/src/lib.rs:960]: Have no idea, are there any convention or registry guiding this?

**Review by HackFisher — CHANGES_REQUESTED** [review]
Need to bump spec_version of crab runtime and darwinia runtime.

Any substrate dep changes or runtime changes need bump spec, normally we will do it always as a good practice for 100% safety concern.

Reference:

https://github.com/darwinia-network/darwinia/pull/586/files#diff-c23cc93db5b2f8c1b19170f6ce738217642392f312028b87011d240406b255b3L175

https://github.com/darwinia-network/darwinia/pull/586/files#diff-3f724ab6074b701b1c5ea4d313e45ab99284f628fda2c91959475baae53e7565L140

**Comment by AsceticBear** [comment]
@HackFisher  I know about this.

> This PR only deals with the tasks that to be completed to add dvm into crab network. As for the crab version upgrade, I will take a separate PR after this modification is approved.

Is this okay?  Separate into two PRs to make review easier.

**Comment by HackFisher** [comment]
> @HackFisher I know about this.
> 
> > This PR only deals with the tasks that to be completed to add dvm into crab network. As for the crab version upgrade, I will take a separate PR after this modification is approved.
> 
> Is this okay? Separate into two PRs to make review easier.

Prefer to make it an atomic PR in case we forget it, this is one of the most severe cases which could cause the network down, so I prefer to prevent it by using atomic PR review convention, not in a separate PR

**Comment by AsceticBear** [comment]
> Need to bump spec_version of crab runtime and darwinia runtime.
> 
> Any substrate dep changes or runtime changes need bump spec, normally we will do it always as a good practice for 100% safety concern.
> 
> Reference:
> 
> https://github.com/darwinia-network/darwinia/pull/586/files#diff-c23cc93db5b2f8c1b19170f6ce738217642392f312028b87011d240406b255b3L175
> 
> https://github.com/darwinia-network/darwinia/pull/586/files#diff-3f724ab6074b701b1c5ea4d313e45ab99284f628fda2c91959475baae53e7565L140

Done

File: runtime/crab/src/lib.rs (hunk 1 above)
Comment (AsceticBear) [reply runtime/crab/src/lib.rs:960]: @AurevoirXavier  Did you know something about this?

**Review by AurevoirXavier — APPROVED** [review]

**Comment by AsceticBear** [comment]
The dvm component is not stable enough yet. Thus, it will not be added to the crab network at present.

Close this PR now.

**Pull request closed without merging**